File: ports/csharp/src/core/io/BcsFile.cs
using System.IO;

namespace BCS.Core.IO {
    // BcsFile abstracts standard file I/O operations for C#
    public class BcsFile {
        private string _path;

        public BcsFile(string path) {
            _path = path;
        }

        public bool Exists() {
            return File.Exists(_path);
        }

        public string ReadAll() {
            return File.ReadAllText(_path);
        }

        public void WriteAll(string payload) {
            File.WriteAllText(_path, payload);
        }
    }
}

File: ports/java/src/core/io/BcsFile.java
package bcs.core.io;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.Path;

// BcsFile abstracts standard file I/O operations for Java
public class BcsFile {
    private final Path path;

    public BcsFile(String pathStr) {
        this.path = Paths.get(pathStr);
    }

    public boolean exists() {
        return Files.isRegularFile(this.path);
    }

    public String readAll() throws IOException {
        return new String(Files.readAllBytes(this.path));
    }

    public void writeAll(String payload) throws IOException {
        Files.write(this.path, payload.getBytes());
    }
}
